dynawo/sources/Models/CPP/Controls/Voltage/ModelVoltageMeasurementsUtilities/DYNModelVoltageMeasurementsUtilities.cpp

```diff
@@ -0,0 +1,442 @@
+    name << "running_" << i << "_value";
+    variables.push_back(VariableNativeFactory::createState(name.str(), BOOLEAN));
+  }
+}
+
+void
+ModelVoltageMeasurementsUtilities::defineParameters(vector<ParameterModeler>& parameters) {
+  parameters.push_back(ParameterModeler("nbInputs", VAR_TYPE_INT, EXTERNAL_PARAMETER));
+  parameters.push_back(ParameterModeler("updateStep", VAR_TYPE_DOUBLE, EXTERNAL_PARAMETER));
+}
+
+void
+ModelVoltageMeasurementsUtilities::setSubModelParameters() {
+  nbConnectedInputs_ = findParameterDynamic("nbInputs").getValue<int>();
+  step_ = findParameterDynamic("updateStep").getValue<double>();
+}
+
+void
+ModelVoltageMeasurementsUtilities::defineElements(std::vector<Element> &elements, std::map<std::string, int>& mapElement) {
+  addElement("min", Element::STRUCTURE, elements, mapElement);
+  addSubElement("value", "min", Element::TERMINAL, name(), modelType(), elements, mapElement);
+  addElement("max", Element::STRUCTURE, elements, mapElement);
+  addSubElement("value", "max", Element::TERMINAL, name(), modelType(), elements, mapElement);
+  addElement("average", Element::STRUCTURE, elements, mapElement);
+  addSubElement("value", "average", Element::TERMINAL, name(), modelType(), elements, mapElement);
+
+  stringstream names;
+  for (size_t i = 0; i < nbConnectedInputs_; ++i) {
+    names.str("");
+    names.clear();
+    names << "UMonitored_" << i;
+    addElement(names.str(), Element::STRUCTURE, elements, mapElement);
+    addSubElement("value", names.str(), Element::TERMINAL, name(), modelType(), elements, mapElement);
+  }
+
+  addElement("tLastUpdate_", Element::STRUCTURE, elements, mapElement);
+  addSubElement("value", "tLastUpdate_", Element::TERMINAL, name(), modelType(), elements, mapElement);
+
+  for (size_t i = 0; i < nbConnectedInputs_; ++i) {
+    names.str("");
+    names.clear();
+    names << "running_" << i;
+    addElement(names.str(), Element::STRUCTURE, elements, mapElement);
+    addSubElement("value", names.str(), Element::TERMINAL, name(), modelType(), elements, mapElement);
+  }
+}
+
+void
+ModelVoltageMeasurementsUtilities::dumpUserReadableElementList(const std::string& /*nameElement*/) const {
+  Trace::info() << DYNLog(ElementNames, name(), modelType()) << Trace::endline;
+  Trace::info() << "  ->" << "UMonitored_" << "<0-" << nbConnectedInputs_ << ">_value" << Trace::endline;
+  Trace::info() << "  ->" << "running_" << "<0-" << nbConnectedInputs_ << ">_value" << Trace::endline;
+}
+
+void
+ModelVoltageMeasurementsUtilities::setFequations() {
+  // setFequations not needed
+}
+
+void
+ModelVoltageMeasurementsUtilities::checkDataCoherence(const double /*t*/) {
+  // Nothing to do in this case.
+}
+
+double
+ModelVoltageMeasurementsUtilities::computeMin(unsigned int &minIdx) const {
+  double minSoFar = std::numeric_limits<double>::max();
+  minIdx = nbConnectedInputs_;
+  for (std::size_t i = 0; i < nbConnectedInputs_; i++) {
+    if (isRunning(i) && (minSoFar > yLocal_[i])) {
```

Comment:
I would propose to add && !doubleEquals(minSoFar, yLocal_[i]) to avoid switching between 2 values if they are identical

Reply:
I thought values would be distinct enough to never be in a case where we would have this case. Even though, the strict inequality could have been enough (or not?) 

But added nonetheless!